### ISSUE TITLE
Problem in ClusterGAN

### ISSUE THREAD

**Issue by Jason-cpp** [eriklindernoren/PyTorch-GAN#75]
The problem is in:
```python
443:     ge_loss = torch.mean(D_gen) + betan * zn_loss + betac * zc_loss
465      d_loss = torch.mean(D_real) - torch.mean(D_gen) + grad_penalty
```
I think it shoud be
```python
443     ge_loss = -torch.mean(D_gen) + betan * zn_loss + betac * zc_loss
465     d_loss = -torch.mean(D_real) + torch.mean(D_gen) + grad_penalty

**Comment by yangjunting100** [comment]
I think you're right. Because that's what wgan's formula is supposed to be. Have you seen the difference between the two? thank you!

**Comment by yangjunting100** [comment]
I did an experiment and found that the two formulas loss converges to almost the same. The quality of image generation is similar.  Wonder if MNIST is too simple or the formula itself??
